src/cloud-api-adaptor/cmd/agent-protocol-forwarder/main.go

```diff
@@ -63,23 +73,32 @@ func (cfg *Config) Setup() (cmd.Starter, error) {
 		flags.StringVar(&tlsConfig.KeyFile, "cert-key", "", "cert key")
 		flags.BoolVar(&tlsConfig.SkipVerify, "tls-skip-verify", false, "Skip TLS certificate verification - use it only for testing")
 		flags.BoolVar(&disableTLS, "disable-tls", false, "Disable TLS encryption - use it only for testing")
+		flags.BoolVar(&secureComms, "secure-comms", false, "Use SSH to secure communication between cluster and peer pods")
+		flags.StringVar(&secureCommsInbounds, "secure-comms-inbounds", "", "Inbound tags for secure communication tunnels")
+		flags.StringVar(&secureCommsOutbounds, "secure-comms-outbounds", "", "Outbound tags for secure communication tunnels")
 	})
 
-	if !disableTLS {
-		cfg.tlsConfig = &tlsConfig
-	}
-
 	cmd.ShowVersion(programName)
 
 	if showVersion {
 		cmd.Exit(0)
 	}
 
-	for path, obj := range map[string]interface{}{
-		cfg.configPath: &cfg.daemonConfig,
-	} {
-		if err := load(path, obj); err != nil {
-			return nil, err
+	if err := load(cfg.configPath, &cfg.daemonConfig); err != nil {
+		return nil, err
+	}
+
+	if secureComms {
+		ppssh.Singleton()
+		cfg.listenAddr = "127.0.0.1:15150"
```

Comment:
OK, I see. I got your point.

I think using a fixed port number will cause a conflict when another process already uses the same port, but it will rarely occur in our use case.

Reply:
My 2 cents is to not bother with this (i.e. ignore the `listenAddr` flag), but if anyone has concern, we can extract the port part of this flag or assert that the flag has a value of 127.0.0.1:<some port>, parse it and use the port...

Reply:
I think extracting the port number from this flag is a good idea. I also think it is convenient for users to understand what is happening to show a warning message when the host name part of this flag is not "localhost" or "127.0.0.1".

The message will be something like this.
```
Warning: listen address is changed to 127.0.0.1:15150 since secure-comms is enabled.
```

Reply:
@yoheiueda 
Pushed a small change to support extracting the port number from this flag and logging when changing the content of the flag (note that we log by default since the default is "0.0.0.0:15150" - so I am removed the "Warning" term from the log entry.